service/tablet_allocator.cc

```diff
@@ -1601,8 +1601,16 @@ class load_balancer {
                 if (skip) {
                     if (src_node_info.drained && skip->viable_targets.empty()) {
                         auto replicas = tmap.get_tablet_info(source_tablet.tablet).replicas;
-                        throw std::runtime_error(fmt::format("Unable to find new replica for tablet {} on {} when draining {} (nodes {}, replicas {})",
-                                                        source_tablet, src, nodes_to_drain, nodes_by_load_dst, replicas));
+                        int number_of_replicas = std::size(replicas);
+                        int number_of_nodes = std::size(nodes);
+                        if (number_of_replicas == number_of_nodes) {
+                            throw std::runtime_error(format("Unable to find new replica for tablet {} on {} when draining {}"
```

Comment:
"nodes" contains the set of nodes to balance, which may be fewer nodes than the number of nodes in a DC. See 5e28bf9b.

Also "replicas" includes replicas in all DCs, and "nodes" is a subset of nodes in the currently balanced DC. So this logic works only when there is 1 DC in the cluster.